frontend/src/features/mapSideBar/tabs/propertyDetails/usePropertyDetails.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  ALR_LAYER_URL,
+  ELECTORAL_LAYER_URL,
+  HWY_DISTRICT_LAYER_URL,
+  INDIAN_RESERVES_LAYER_URL,
+  IUserLayerQuery,
+  MOTI_REGION_LAYER_URL,
+  useLayerQuery,
+} from 'components/maps/leaflet/LayerPopup';
+import { useProperties } from 'hooks';
+import useIsMounted from 'hooks/useIsMounted';
+import { LatLngLiteral } from 'leaflet';
+import { useEffect, useMemo, useState } from 'react';
+import { pidFormatter } from 'utils';
+
+import { IPropertyDetailsForm, toFormValues } from './PropertyDetailsTabView.helpers';
+
+export function usePropertyDetails(pid?: string) {
```

Comment:
I was hoping this method would be general enough to just return a Property from the api. Would it be possible to make this file return that instead? I can refactor that if necessary

Reply:
I'm merging right now. You can refactor if needed

Reply:
and it does return a property from the api - not sure I follow

Reply:
if you just need an API_Property instance then use `const { getPropertyWithPid } = useProperties();`